packages/core/content-manager/server/src/controllers/collection-types.ts

```diff
@@ -422,36 +422,35 @@ export default {
       return ctx.forbidden();
     }
 
-    const permissionQuery = await permissionChecker.sanitizedQuery.publish(ctx.query);
-    const populate = await getService('populate-builder')(model)
-      .populateFromQuery(permissionQuery)
-      .populateDeep(Infinity)
-      .countRelations()
-      .build();
+    const { locale } = getDocumentLocaleAndStatus(body);
 
-    const entityPromises = ids.map((id: any) => documentManager.findOne(id, model, { populate }));
-    const entities = await Promise.all(entityPromises);
+    const documentLocales = await documentManager.findLocales(documentIds, model, {
+      locale,
+    });
 
-    for (const entity of entities) {
-      if (!entity) {
-        return ctx.notFound();
-      }
+    if (documentLocales.length === 0) {
+      return ctx.notFound();
+    }
 
-      if (permissionChecker.cannot.publish(entity)) {
+    for (const document of documentLocales) {
+      if (permissionChecker.cannot.publish(document)) {
         return ctx.forbidden();
       }
     }
 
```

Comment:
I think the .cannot.publish check should only be done with the draft versions and I believe you might have draft and published versions here right?

Reply:
you might as well not query those one when looking for the find locales 🤔

Reply:
and same for the unpublish action, you should only check published documents